src/Model/Subsite.php

```diff
@@ -885,6 +885,16 @@ public function domain()
      */
     public function getPrimarySubsiteDomain()
     {
+        // Main site
+        if (!$this->isInDB()) {
```

Comment:
Now seeing the method this is in and the not-in-db check... is there a better/different place this could go to? Alternatively, can we get more explanation why it's here and why we need to check it's not in the db?

Reply:
An alternative would be `if (!$this->ID) {`.
This is for situations like `SiteTreeSubsites::alternateAbsoluteLink` (inline below for convenience) where we call `domain` (which calls `getPrimarySubsiteDomain`) on the relationship directly from the `has_one` relation object.
If the page has no `SubiteID` (i.e. it belongs to the main site), the relation object will still be an instantiated `Subsite` object which has no ID and which returns `false` from `isInDB` - so we end up getting the main site domain.

I can't think of somewhere else to put the call, since the intention is to distinguish between a main site or subsite `Subsite` object (where a main site `Subsite` object is any `Subsite` object without an ID, effectively) when fetching the domain.

This allows project-specific code or code from other modules to remain the same (if they don't have the "if there is a subsite ID" check in place - in that case there's still only a very minor change required) and now get the added benefit of automagically fetching the correct main site domain as well.

https://github.com/silverstripe/silverstripe-subsites/blob/0b7a63e0482a8e004ef34595233654124189fcbe/src/Extensions/SiteTreeSubsites.php#L410-L417

Reply:
Just realised I didn't explicitly say why we need to check for the main site vs an actual subsite there - the main site domains aren't stored on the Domains relation of any subsite object, so we can't just use `$this->Domains()` for the main site.